pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -753,3 +758,58 @@ func getAlertFolderNameFromDashboard(dash *dashboard) string {
 	}
 	return fmt.Sprintf(DASHBOARD_FOLDER, title, dash.Uid) // include UID to the name to avoid collision
 }
+
+// CreateDefaultFoldersForAlertingMigration creates a folder dedicated for alerting if no folders exist
+func CreateDefaultFoldersForAlertingMigration(mg *migrator.Migrator) {
+	if !mg.Cfg.UnifiedAlerting.IsEnabled() {
+		return
+	}
+	mg.AddMigration("create default alerting folders", &createDefaultFoldersForAlertingMigration{})
+}
+
+type createDefaultFoldersForAlertingMigration struct {
+	migrator.MigrationBase
+}
+
+func (c createDefaultFoldersForAlertingMigration) Exec(sess *xorm.Session, migrator *migrator.Migrator) error {
+	helper := folderHelper{
+		sess: sess,
+		mg:   migrator,
+	}
+
+	var rows []struct {
+		Id   int64
+		Name string
+	}
+
+	if err := sess.Table("org").Cols("id", "name").Find(&rows); err != nil {
+		return fmt.Errorf("failed to read the list of organizations: %w", err)
+	}
+
+	numberFoldersPerOrg, err := helper.getOrgsThatHaveFolders()
+	if err != nil {
+		return fmt.Errorf("failed to count folders per organization: %w", err)
```

Comment:
Same than above, AFAIU the method doesn't count the number of folders per org, but the orgs with folders. Maybe that was the original purpose of the function but then it you changed it to better fit our needs?

Reply:
It used to be count but then I simplified it even more